pkg/shares/shares.go

```diff
@@ -191,6 +191,51 @@ func (s *Share) rawDataStartIndex() int {
 	return index
 }
 
+// RawDataWithReserved returns the raw share data while taking reserved bytes into account.
+func (s *Share) RawDataUsingReserved() (rawData []byte, err error) {
+	rawDataStartIndexUsingReserved, err := s.rawDataStartIndexUsingReserved()
+	if err != nil {
+		return nil, err
+	}
+
+	// This means share is the last share and does not have any transaction beginning in it
+	if rawDataStartIndexUsingReserved == 0 {
+		return []byte{}, nil
+	}
```

Comment:
[question] I expect this conditional to evaluate to `true` for the last share in the transaction namespace. However, for that share, I would expect that `RawDataUsingReserved` returns the raw data from the last share (i.e. the remainder of the last transaction that overflowed into the last share).

```go
func TestRawDataUsingReserved(t *testing.T) {
	type testCase struct {
		name  string
		share Share
		want  []byte
	}
	infoByte, err := NewInfoByte(appconsts.ShareVersionZero, false)
	assert.NoError(t, err)
	reservedBytes := []byte{0, 0, 0, 0}
	lastTxRemainingBytes := []byte{1, 2, 3}

	lastTransactionShareData := []byte{}
	lastTransactionShareData = append(lastTransactionShareData, appns.TxNamespace.Bytes()...)
	lastTransactionShareData = append(lastTransactionShareData, byte(infoByte))
	lastTransactionShareData = append(lastTransactionShareData, reservedBytes...)
	lastTransactionShareData = append(lastTransactionShareData, lastTxRemainingBytes...)

	testCases := []testCase{
		{
			name: "last transaction share",
			share: Share{
				data: lastTransactionShareData,
			},
			want: lastTxRemainingBytes,
		},
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			rawData, err := tc.share.RawDataUsingReserved()
			require.NoError(t, err)
			assert.Equal(t, tc.want, rawData)
		})
	}
}
```

results in

```
            	            	expected: []byte{0x1, 0x2, 0x3}
            	            	actual  : []byte{}
```

Reply:
Since `RawDataUsingReserved` is only called for the first share when parsing out of context shares, in the case when the share only contains remainder of the last transaction that overflowed into this share, there are no complete transactions, so returning an empty byte array, in this case, would make sense.